server.mjs does not wait for listen() before test runner starts

The Node test server starts asynchronously, but `test-engine/server/server.mjs` currently calls `server.listen(port)` without exposing or awaiting readiness.

In consumers that start the server via side-effect import, for example:

```js
await import('./test-engine/server/server.mjs')
await runTests(...)
```

the import can resolve before the server has emitted `listening`. On slower platforms, especially Windows CI, the first client request can race server startup and fail intermittently.

Observed symptoms:

```console
SERVER WARNING: State not found for <uuid>
Error: Process completed with exit code 1
```

The likely sequence is:

1. Worker imports `server.mjs`.
2. `server.listen(port)` is called, but the server is not ready yet.
3. Test setup sends `PUT /config/<uuid>`.
4. The request can fail before the server accepts connections.
5. The harness continues to later `GET /state/<uuid>`.
6. Since no config/state was created, the server logs `State not found`.

Suggested fix:

```js
await new Promise((resolve, reject) => {
  server.once('listening', resolve)
  server.once('error', reject)
  server.listen(port)
})
```

This preserves the existing side-effect import behavior while making `await import('./server.mjs')` mean the server is actually listening before tests proceed.

This was found while running Undici’s cache interceptor test harness against the vendored `cache-tests` submodule in https://github.com/nodejs/undici/issues/5199

--- a/test-engine/server/server.mjs
+++ b/test-engine/server/server.mjs
@@ -51,4 +51,13 @@ server.on('listening', () => {
     : server.address().address
   console.log(`Listening on ${protocol.toLowerCase()}://${host}:${server.address().port}/`)
 })
-server.listen(port)
+await new Promise((resolve, reject) => {
+  function onListening () {
+    server.off('error', reject)
+    resolve()
+  }
+
+  server.once('listening', onListening)
+  server.once('error', reject)
+  server.listen(port)
+})

